corehq/apps/api/resources/v0_5.py

```diff
@@ -285,7 +288,7 @@ def obj_update(self, bundle, **kwargs):
     def obj_delete(self, bundle, **kwargs):
         user = CommCareUser.get(kwargs['pk'])
         if user:
-            user.retire()
+            user.retire(deleted_by=bundle.request.user, deleted_via=USER_CHANGE_VIA_API)
```

Comment:
@mkangia the WebUserResources also needs something here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/api/resources/v0_5.py#L382

I made an attempt at this but had trouble getting the tests to pass.

Reply:
Thanks for sharing that @snopoke .

Isn't it raising another exception for missing argument `deleted_by` https://github.com/dimagi/commcare-hq/blob/e03d3f2000a8635c83e1b81cfefa06f7b1881fba/corehq/apps/users/models.py#L1179 ?

Reply:
yes, that was the exception though it wasn't straight forward to just add that there since there were other issues after that, notably that the Django user did not exists for some reason. Perhaps due to transaction management in the request?

Reply:
The user is getting deleted right after its creation due to an error, do we really need to track that?
i am thinking if we should just skip auditing that. 
Its very much possible the django user isn't created because an exception happened even before it could be added and hence we are now just cleaning the couch record.

Reply:
Since the creation of the user happens in multiple steps it would be possible for the user to get created and saved before the error which would result in a log entry for the creation but not one for the delete.

Either way, this will 500 if it ever hits this code.

Reply:
> which would result in a log entry for the creation but not one for the delete

Ah I see, ya that would be confusing. 
So, it would be hard to be sure that there will be a django user when that delete is called, hence `deleted_by` could be None. 
Is there a PR you already have started on or want me to take a look ?

Reply:
I blew away my changes since they clearly weren't working.